scripts/seed-database/upsert-production-flows.js

```diff
@@ -50,65 +60,197 @@ const LOCAL_GRAPHQL_ADMIN_SECRET = process.env.HASURA_GRAPHQL_ADMIN_SECRET;
       }
     `);
 
-    await localClient.request(`
-      mutation DeleteTeams(
-        $teams_slugs: [String!]!, 
-      ) {
-        delete_teams(where: {slug: {_in: $teams_slugs}}) {
-          affected_rows
+    const {
+      teams: localTeams,
+      flows: localFlows,
+    } = await localClient.request(`
+      query GetAllFlowsAndTeams {
+        flows {
+          id
+          slug
+          team_id
+        }
+        teams {
+          id
+          slug
         }
       }
-      `,
+    `);
+
+    // Teams and Flows have 2 unique fields: `id` and `slug`, but the upsert `on_conflict` query only allows us to check for one of them. 
+    // So if both fields exist in the current database, the query will fail. 
+    // To prevent errors, we upsert production data in two different queries, filtering them by `id` and `slug`.
+
+    const localTeamsSlugs = localTeams.map(team => team.slug);
+    const {
+      false: teamsToUpsertById,
+      true: teamsToUpsertBySlug
+    } = groupBy(productionTeams, (team) => localTeamsSlugs.includes(team.slug));
+
+    const {
+      false: flowsToUpsertById,
+      true: flowsToUpsertBySlug
+    } = groupBy(
+      productionFlows.map(flow => ({ ...flow, version: 1, })),
```

Comment:
:+1: this is clever and cleaner than querying the most recent operation per flow! is it worth having a brief code comment here too as a reminder that that flow version is overwritten to match the operation version?